[wip] ci/soak test

hlolli opened the pull request:
I'm going to be working on these tests for a bit more, and see how travis-ci reacts to them.

With python unittest I'm able to run ~740 csound renders and checksum checks in under 90 seconds.

seppeljordan commented on tests/soak/test_checksum.py:
```diff
@@ -43,9 +44,6 @@
 if os.path.exists(travis_csound):
```
Put this in def setUp(self): method

seppeljordan replied:
nevermind...

seppeljordan commented on tests/soak/test_checksum.py:
```diff
@@ -67,16 +65,17 @@ class SoakOpcodes(unittest.TestCase):
     @parameterized.expand(test_data_parameterized # , name_func=custom_name_func
     )
```
```python
def setUp(self):
    os.makedirs(os.path.join(os.getenv("TRAVIS_BUILD_DIR"), "build"), exist_okay=True)
```

hlolli replied:
I don't understand, why would it make sense to create $TRAVIS_BUILD_DIR/build, you mean setUp a single temporary directory?

seppeljordan replied:
my bad.

```
def setUp(self):
    os.makedirs("tmp", exist_okay=True)

hlolli replied:
```
os.makedirs("tmp", exist_ok=True)
```

did it. But adding method to the class, causes no test to run whatsoever. So I did it like this

```diff
@@ -62,20 +61,19 @@ def custom_name_func(testcase_func, index, params):
     )
 
 class SoakOpcodes(unittest.TestCase):
-    @parameterized.expand(test_data_parameterized # , name_func=custom_name_func
-    )
+    @parameterized.expand(test_data_parameterized , name_func=custom_name_func)
     def test(self, path, expected):
-        with tempfile.TemporaryDirectory() as tmpdir:
-            name = os.path.basename(path).replace(".csd.json", "")
-            csd = path.replace(".json", "")
-            rendered_file_path = os.path.join(tmpdir, name + ".wav")
-            subprocess.run(
-                [csound_path, csd, "-o", rendered_file_path] + cli_options,
-                check=True, env=env
-            )
-            with open(rendered_file_path, "rb") as handle:
-                binary_data = handle.read()
-            hasher = sha256()
-            hasher.update(binary_data)
-            actual_hashsum = hasher.hexdigest()
-            self.assertEqual(actual_hashsum, expected)
+        os.makedirs("tmp", exist_ok=True)
+        name = os.path.basename(path).replace(".csd.json", "")
+        csd = path.replace(".json", "")
+        rendered_file_path = os.path.join("tmp", name + ".wav")
+        subprocess.run(
+            [csound_path, csd, "-o", rendered_file_path] + cli_options,
+            check=True, env=env
+        )
+        with open(rendered_file_path, "rb") as handle:
+            binary_data = handle.read()
+        hasher = sha256()
+        hasher.update(binary_data)
+        actual_hashsum = hasher.hexdigest()
+        self.assertEqual(actual_hashsum, expected)
```

hlolli commented:
Could you review this pull request @kunstmusik and @jpffitch 

I wrote a README.md, I moved some of @jpffitch original python code into backup directory and changed the .travis.yaml a bit. The dryruns aren't doing anything right now, but writing print based test for these would be my next step. But merge-ing this will increase our test coverage by a substantial amount.

kunstmusik commented:
@hlolli is runtests.py used anymore? I saw mention there of commandline md5sum vs. the python hashlib code.  It seems that it is still the command used in CMakeLists.txt which is run from "make soak".  

I cancelled the pending Appveyor build except the last one since each one takes a very long time to run. Will wait to see what the result is there.

hlolli commented:
I'll remove the command, don't want to delete the pyhton code from @jpffitch without thumbs up. I'll delete it, since it's doing the same, compare hashes, and we can always revert it again.

kunstmusik commented:
@hlolli Could you update CmakeFiles.txt to call the new command? Would be good to run it here locally using ```make soak``` to test it.

hlolli commented:
Done, I removed the old files, and now make soak should handle both travis and local development. Given that pyhton3 is installed (with pip3).

Let me know if you have any checksum conflicts, my computer is having 17 fails compared to the travis (which makes 675 correct matches). I think it makes sense to keep the reference for now on the CI and research why 2 machines produce 2 different results on these particular opcodes.

Also gen28, space_quad and spdist were segfaulting on the CI so I moved them to the broken directory. I'm going to run them in a docker image of ubuntu to get to the bottom of that (sometime soon)

kunstmusik commented:
I ran here on MacOS and got:

----------------------------------------------------------------------                                      
Ran 691 tests in 167.580s                                                                                   
                                                                                                            
FAILED (failures=12, errors=1)         

Is this expected?

hlolli commented:
TLDR; can you send me the full log report just from the test (pastebin link)?

That error is very unexpected, which test had error?

I'm less surprised by the failures, since if this includes wguide1 and wguide2, I'm also getting different hashes. Why that's the case, is a mystery to me. But given what we know about wav binaries, is that for the same output format, the headers should be exactly the same, so the content must differ in these 12 cases. In in my opinion, there's something wrong happening. (also given that you had ~670 successes).

kunstmusik commented:
@hlolli see https://pastebin.com/dELMbkAC

hlolli commented:
`ERROR: test_opcodes_chuap_chuap_csd_json` would be opcodes/chuap/chuap.csd. Could you run the following file from the csound build and tell me what happens (I suspect you're getting segfault).

```Csound
<CsoundSynthesizer>
<CsInstruments>

sr     = 44100
ksmps  = 32
nchnls = 2
0dbfs  = 1

gibuzztable ftgen 1, 0, 16384, 10, 1

instr 1
; sys_variables = system_vars(5:12); % L,R0,C2,G,Ga,Gb,E,C1 or p8:p15
; integ_variables = [system_vars(14:16),system_vars(1:2)]; % x0,y0,z0,dataset_size,step_size or p17:p19, p4:p5
istep_size    =       p5
iL            =       p8
iR0           =       p9
iC2           =       p10
iG            =       p11
iGa           =       p12
iGb           =       p13
iE            =       p14
iC1           =       p15
iI3           =       p17
iV2           =       p18
iV1           =       p19
iattack       =       0.02
isustain      =       p3
irelease      =       0.02
p3            =       iattack + isustain + irelease
iscale        =       1.0
adamping      linseg  0.0, iattack, iscale, isustain, iscale, irelease, 0.0
aguide        buzz    0.5, 440, sr/440, gibuzztable
aI3, aV2, aV1 chuap   iL, iR0, iC2, iG, iGa, iGb, iE, iC1, iI3, iV2, iV1, istep_size
asignal       balance aV2, aguide
              outs    adamping * asignal, adamping * asignal
endin
</CsInstruments>
<CsScore>
;        Adapted from ABC++ MATLAB example data.
i 1 0 2 1500 .1   -1 -1 -0.00707925 0.00001647 100  1 -.99955324 -1.00028375 1 -.00222159 204.8 -2.36201596260071 3.08917625807226e-03 3.87075614929199 7 .4 .004 1 86 30; torus attractor ( gallery of attractors )
i 1 + 2 1500 .425  0 -1  1.3506168  0              -4.50746268737 -1 2.4924 .93 1 1 0 -22.28662665 .009506608 -22.2861576 32 10 2 20 86 30 ; heteroclinic orbit
i 1 + 2 1024 .05  -1 -1  0.00667    0.000651    10 -1 .856 1.1 1 .06 51.2 -20.200590133667 .172539323568344 -4.07686233520508 2.5 10 .2 1 66 81 ; periodic attractor (torus breakdown route)
i 1 + 2 1024 0.05 -1 -1 0.00667 0.000651 10 -1 0.856 1.1 1 0.1 153.6 21.12496758 0.03001749 0.515828669 2.5 10 0.2 1 66 81  ; torus attractor (torus breakdown route)'
</CsScore>
</CsoundSynthesizer>
```

Then there are the other less surprising failures, if I take one example, the fmmetal opcode. Then on my computer I'm getting the following failure

```
- cc4c9f01a38fbf8299b7fc3be3accaa13a502e070e7980f161996276f1f14590
+ ed77cca0f6c426bef240ae2c84e94846a689a7a8143d1c1deecb8fd6b84d7ba7
```

and you got

```
- d6956f709a1bdeebe1cb3358a688a642c08ab8a25ed09245224d68f819573cb6
+ ed77cca0f6c426bef240ae2c84e94846a689a7a8143d1c1deecb8fd6b84d7ba7
```

which makes 3 different results on 3 different computers.

kunstmusik commented:
A couple things:

1. error: syntax error, unexpected T_IDENT  (token "chuap") from file opcodes/chuap/chuap.csd (1)
 line 33:
>>>aI3, aV2, aV1 chuap <<<

This is because on my system:

-- EIGEN3_INCLUDE_PATH="EIGEN3_INCLUDE_PATH-NOTFOUND", so disabling BUILD_CHUA_OPCODES                        
-- BUILD_CHUA_OPCODES is disabled.   

2. The soak tests seem to use the system-installed Csound. See tests/commandline/test.py for how it was done to use the version of Csound that was compiled in the build tree. 

3. Where do the soak tests store previously written data?

hlolli commented:
1. ok so I wont worry about that
2. Travis is calling it from the build directory, but passing the location from the cmake makes more sense for both local and ci testing (will change)
3. the .wav files are written into a tmp directory in test/soak/tmp

hlolli commented:
I was able to reproduce the fails in a docker image, errors aside which I didn't install, namely chuap and imageopcodes. The good news is that the hashes are consistent between platforms, the bad news is that the platforms are having effect on reproduce-ability. Will research further...

```Dockerfile
FROM ubuntu:disco-20191011
MAINTAINER Hlöðver Sigurðsson <hlolli@gmail.com>
RUN apt-get update && apt-get install -y locales && rm -rf /var/lib/apt/lists/* \
    && localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
ENV LANG en_US.utf8
RUN apt-get update
RUN apt-get install -y software-properties-common curl alsa-utils cmake git build-essential libsndfile1-dev libasound2-dev bison flex python3-venv
RUN git clone https://github.com/hlolli/csound
RUN cd csound && git checkout c101957d0c5cf4ed637f991eec801bb1fd7b4d28
RUN mkdir -p csound/build
RUN cd csound/build && cmake ../ -DBUILD_PYTHON_OPCODES=0
RUN cd csound/build && make -j4
RUN cd csound/tests/soak && git clone https://github.com/wolever/parameterized.git && python3.7 -m venv "$HOME/venv" && "$HOME/venv"/bin/pip3.7 install wheel && "$HOME/venv"/bin/pip3.7 install ./parameterized
RUN export CSOUND_BIN="/csound/build/csound" && export OPCODE6DIR64="/csound/build" && cd csound/tests/soak && "$HOME/venv"/bin/python test_prepare.py && "$HOME/venv"/bin/python -m unittest -v test_checksum
```

```
======================================================================
FAIL: test_opcodes_adsyn_adsyn_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: 'b3a2d7192505689fc9720fc4d1295b33cd4a6383b63c7f5f0550ae0e6e6b2a8c' != '08a97be195c7e6353bf9bb66c032f67d1b23dea2a00d6a7abb74db8ec6970930'
- b3a2d7192505689fc9720fc4d1295b33cd4a6383b63c7f5f0550ae0e6e6b2a8c
+ 08a97be195c7e6353bf9bb66c032f67d1b23dea2a00d6a7abb74db8ec6970930


======================================================================
FAIL: test_opcodes_fmmetal_fmmetal_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '6394e76182805f86bf107df12ec0cc11f4f817c4b418261062e4c933d4c4a4e2' != '2769a83f7fe814410b2c006f454f6d1ab81fb3221cf5203459a9bcbafb90d900'
- 6394e76182805f86bf107df12ec0cc11f4f817c4b418261062e4c933d4c4a4e2
+ 2769a83f7fe814410b2c006f454f6d1ab81fb3221cf5203459a9bcbafb90d900


======================================================================
FAIL: test_opcodes_fmrhode_fmrhode_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: 'cef1fd51a94890042bfa113cfc40ff1ee91b4a5e46c46ee887a9829e0bbfd975' != '1b664725ff7de816d60ee25a8bdc4001c17775e5ef2335472668b2a125bd7691'
- cef1fd51a94890042bfa113cfc40ff1ee91b4a5e46c46ee887a9829e0bbfd975
+ 1b664725ff7de816d60ee25a8bdc4001c17775e5ef2335472668b2a125bd7691


======================================================================
FAIL: test_opcodes_fmwurlie_fmwurlie_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: 'af1777e39717c9655b44fd8614eed1eff27b8729f8b07a9d425f5f771753a7da' != '318d80ecb29a5a353f3ee73953dd6e57978fefcd0b4581e4f09f14743dcdc8ba'
- af1777e39717c9655b44fd8614eed1eff27b8729f8b07a9d425f5f771753a7da
+ 318d80ecb29a5a353f3ee73953dd6e57978fefcd0b4581e4f09f14743dcdc8ba


======================================================================
FAIL: test_opcodes_gen12_gen12_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '33c32e5541d07a219727f0a9ba049a620474bfe614c3207a705f2c0db785f9a8' != '72634709d5cf0f64169273dd0ce18a8b96aa65b630b3e3619643bcda0e86d0bb'
- 33c32e5541d07a219727f0a9ba049a620474bfe614c3207a705f2c0db785f9a8
+ 72634709d5cf0f64169273dd0ce18a8b96aa65b630b3e3619643bcda0e86d0bb


======================================================================
FAIL: test_opcodes_lorenz_lorenz_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '53175b0e6206edf61dcec7259614c7d37c99819181a2af4cdb38d03939ad7e6e' != '20fb9c939cd7f24bba46aec4a4825cfe202dc5b157d28d897d53ee5eac2a539c'
- 53175b0e6206edf61dcec7259614c7d37c99819181a2af4cdb38d03939ad7e6e
+ 20fb9c939cd7f24bba46aec4a4825cfe202dc5b157d28d897d53ee5eac2a539c


======================================================================
FAIL: test_opcodes_lpreson_lpreson_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '49b1b240aaea9cce844e32164f1a1fa6ffe77c66f3980a093ebff6bc0f5fc3a2' != '5715e5959057a574838a719586bfaad87940d6e44fc4b691e7060b97db08dd15'
- 49b1b240aaea9cce844e32164f1a1fa6ffe77c66f3980a093ebff6bc0f5fc3a2
+ 5715e5959057a574838a719586bfaad87940d6e44fc4b691e7060b97db08dd15


======================================================================
FAIL: test_opcodes_mincer_mincer_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '386f50e070a3c7e86a497f43bd7b58bb74cad67107620f042e70db4b71c7d783' != 'b8b1619493fce5ac5440ea19ab59258653d06903d61538d47a50d8a0e2e519de'
- 386f50e070a3c7e86a497f43bd7b58bb74cad67107620f042e70db4b71c7d783
+ b8b1619493fce5ac5440ea19ab59258653d06903d61538d47a50d8a0e2e519de


======================================================================
FAIL: test_opcodes_phaser2_phaser2_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '8ad1858356cf4d17e1907b5d4a6914712d07b70372a1218bb02a69d1ad99b38a' != 'dcfc461fdb1103168d190726933c52f85902dc1af4d410f778efe9707485e685'
- 8ad1858356cf4d17e1907b5d4a6914712d07b70372a1218bb02a69d1ad99b38a
+ dcfc461fdb1103168d190726933c52f85902dc1af4d410f778efe9707485e685


======================================================================
FAIL: test_opcodes_plltrack_plltrack_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '846ff789e5b081e33db3aa67db809b32da9e84a4f47f652458f1f64cdea1db74' != 'd07be95e7ada7169a78c4a3fd88dd1f2f8547c2a4c16fe33e00096101da60d1f'
- 846ff789e5b081e33db3aa67db809b32da9e84a4f47f652458f1f64cdea1db74
+ d07be95e7ada7169a78c4a3fd88dd1f2f8547c2a4c16fe33e00096101da60d1f


======================================================================
FAIL: test_opcodes_pvsvoc_pvsvoc_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '38f2f644524d6afdc91fe44f62b0ed4e9117a6e7db90e565813482b2ea3c7236' != 'd01915337760f81fc08750c20e84d5a2ceb650d81e2a3c26f8bc692ef14e119e'
- 38f2f644524d6afdc91fe44f62b0ed4e9117a6e7db90e565813482b2ea3c7236
+ d01915337760f81fc08750c20e84d5a2ceb650d81e2a3c26f8bc692ef14e119e


======================================================================
FAIL: test_opcodes_spat3d_spat3d_UHJ_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '775437e28d8ba5648ad1c33505d7270b1fbce744468c6cf83c89fed7184e0d53' != '8a218cf4375290ba167048fd74317dbfc169cbb53c2af00cade5a5f5c7c5d12e'
- 775437e28d8ba5648ad1c33505d7270b1fbce744468c6cf83c89fed7184e0d53
+ 8a218cf4375290ba167048fd74317dbfc169cbb53c2af00cade5a5f5c7c5d12e


======================================================================
FAIL: test_opcodes_tbvcf_tbvcf_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '6b84b6fe99ee1772fd9fd6682510d9b29244dfcceab89eeae59a9b0af701b6bd' != '9137bf3d95ea5aa7c575019e1792a3f65e2fc5db654d388c29b1e76c8fbe53fc'
- 6b84b6fe99ee1772fd9fd6682510d9b29244dfcceab89eeae59a9b0af701b6bd
+ 9137bf3d95ea5aa7c575019e1792a3f65e2fc5db654d388c29b1e76c8fbe53fc


======================================================================
FAIL: test_opcodes_temposcal_temposcal_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: 'e91e602e5df680b513f4f56104c6029eb7f4dfed0f165e4ad60cacbfbce5a46d' != '79b19889e53d265a6035a4d16a74724735cba41156b7b95a9d8611028ff9f80e'
- e91e602e5df680b513f4f56104c6029eb7f4dfed0f165e4ad60cacbfbce5a46d
+ 79b19889e53d265a6035a4d16a74724735cba41156b7b95a9d8611028ff9f80e


======================================================================
FAIL: test_opcodes_wgbrass_wgbrass_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: 'a01a2592acecaa8bfb6998bbf7640b72e22eb525a47f8e3406abc4458c62da4a' != '7d2b746012a669b82e3a9e77ddf9aed285dfd178cf066415ae5cd7715cb33805'
- a01a2592acecaa8bfb6998bbf7640b72e22eb525a47f8e3406abc4458c62da4a
+ 7d2b746012a669b82e3a9e77ddf9aed285dfd178cf066415ae5cd7715cb33805


======================================================================
FAIL: test_opcodes_wguide2_wguide2_csd_json (test_checksum.SoakOpcodes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/venv/lib/python3.7/site-packages/parameterized/parameterized.py", line 518, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/csound/tests/soak/test_checksum.py", line 77, in test
    self.assertEqual(actual_hashsum, expected)
AssertionError: '7427ded188c4a74845f555fb293097b661125aae04e02cf6e21f4d1143f2ac4a' != '455d58b3ebec0e655a6ef1749103455c7964cbffc19f192acfea5ee63aaf731e'
- 7427ded188c4a74845f555fb293097b661125aae04e02cf6e21f4d1143f2ac4a
+ 455d58b3ebec0e655a6ef1749103455c7964cbffc19f192acfea5ee63aaf731e


----------------------------------------------------------------------
Ran 691 tests in 129.667s

FAILED (failures=16, errors=4)
```

hlolli commented:
This afternoon I had exactly the same failures on my docker image as Travis had (both are Ubuntu based). Now I have 100 differences to the Travis build but only 13 between my computer and the docker image, running the exact same code. Before I loose my sanity, I'm going to finish this at another time.

hlolli commented:
I'm still unsure about this PR, the non reproduceability when generating binary sound files was a huge headache which I'm not sure I'll have the time for. Using some sandboxed test-first c compiler could be a solution. Closing this PR but keeping my branch open.
